react-native-gutenberg-bridge/android/src/main/java/org/wordpress/mobile/WPAndroidGlue/WPAndroidGlueCode.java

```diff
@@ -681,25 +681,24 @@ public void appendUploadMediaFiles(ArrayList<Media> mediaList) {
         } else {
             // This case is for media that is shared from the device
             for (Media mediaToAppend : mediaList) {
-                sendOrDeferAppendMediaSignal(mediaToAppend.getId(), mediaToAppend.getUrl(),
-                        mediaToAppend.getType(), mediaToAppend.getCaption());
+                sendOrDeferAppendMediaSignal(mediaToAppend);
             }
         }
 
         mAppendsMultipleSelectedToSiblingBlocks = false;
     }
 
-    private void sendOrDeferAppendMediaSignal(final int mediaId, final String mediaUri, final String mediaType, final String caption) {
+    private void sendOrDeferAppendMediaSignal(Media media) {
         // if editor is mounted, let's append the media file
         if (mIsEditorMounted) {
-            if (!TextUtils.isEmpty(mediaUri) && mediaId > 0) {
+            if (!TextUtils.isEmpty(media.getUrl()) && media.getId() > 0) {
                 // send signal to JS
-                appendNewMediaBlock(mediaId, mediaUri, mediaType);
+                appendNewMediaBlock(media.getId(), media.getUrl(), media.getType());
             }
         } else {
             // save the URL, we'll add it once Editor is mounted
             synchronized (WPAndroidGlueCode.this) {
-                mMediaToAddAfterMounting.put(mediaId, new Media(mediaId, mediaUri, mediaType, caption));
+                mMediaToAddAfterMounting.put(media.getId(), media);
```

Comment:
We shouldn't need to create a new object now that the `Media` is an immutable data class.